central/declarativeconfig/singleton.go

```diff
@@ -10,10 +18,30 @@ var (
 	instance Manager
 )
 
+// ReconciliationErrorReporter processes declarative resources reconciliation errors.
+type ReconciliationErrorReporter interface {
+	ProcessError(protoValue proto.Message, err error)
+}
+
+type noOpErrorReporter struct{}
+
+func (n noOpErrorReporter) ProcessError(_ proto.Message, _ error) {}
+
 // ManagerSingleton provides the instance of Manager to use.
-func ManagerSingleton() Manager {
+func ManagerSingleton(registry authproviders.Registry) Manager {
 	once.Do(func() {
-		instance = New(env.DeclarativeConfigReconcileInterval.DurationSetting(), env.DeclarativeConfigWatchInterval.DurationSetting())
+		instance = New(
+			env.DeclarativeConfigReconcileInterval.DurationSetting(),
+			env.DeclarativeConfigWatchInterval.DurationSetting(),
+			map[reflect.Type]updater.ResourceUpdater{
+				authProviderType:  updater.NewAuthProviderUpdater(authProviderDatastore.Singleton(), registry),
+				groupType:         updater.NewGroupUpdater(groupDataStore.Singleton()),
+				roleType:          updater.NewRoleUpdater(roleDatastore.Singleton()),
+				permissionSetType: updater.NewPermissionSetUpdater(roleDatastore.Singleton()),
+				accessScopeType:   updater.NewAccessScopeUpdater(roleDatastore.Singleton()),
+			},
```

Comment:
How about moving this map initialization call to the `updater` package, in a `DefaultResourceUpdater()` function or something like that?

It's more of a nit, but right now the manager has imports it shouldn't necessarily require / know (talking about the datastore dependencies).